src/earn/poolInfo.ts

```diff
@@ -32,3 +36,47 @@ export async function fetchAavePoolInfo(assetAddress: Address) {
     throw err
   }
 }
+
+/** @beta - Exclude from Knip dep check */
+export async function fetchAavePoolUserBalance({
+  assetAddress,
+  walletAddress,
+}: {
+  assetAddress: Address
+  walletAddress: Address
+}) {
+  try {
+    if (!assetAddress) {
```

Comment:
would we expect this to happen with a `viem` `Address`?

Reply:
It was mainly so the usage of hooks can handle possible undefined values for example: `const asyncPoolInfo = useAavePoolInfo(token?.address as Address)`. Since hooks are coming in a later PR this can be removed, but will likely be needed when hooks are added.

Reply:
The type should reflect this if it can be potentially undefined. I don't think we should be doing `token?.address as Address` anywhere. It can lead to nasty bugs down the line, if the function doesn't handle it being potentially nullable

Reply:
seems worth removing since it's confusing given the function signature/typing

Reply:
I was going to do something like this to get the pool info in the component:

```TypeScript
  const token = useTokenInfo(networkConfig.arbUsdcTokenId)
  const walletAddress = useSelector(walletAddressSelector)
  const asyncPoolInfo = useAavePoolInfo(token?.address as Address)
  const asyncPoolUserInfo = useAavePoolUserBalance({
    assetAddress: token?.address as Address,
    walletAddress: walletAddress as Address
  })
```